content/doc/book/pipeline/pipeline-best-practices.adoc

```diff
@@ -94,15 +94,41 @@ If building containers will not work, disable concurrency on the Pipeline or use
 
 **Also, be aware that both of these methods have slower time to results of builds than using unique resources for each job**
 
-=== Using @NonCPS
+=== Avoiding `NotSerializableException` 
 
-Pipeline code is CPS-transformed so that Pipelines are able to resume after a Jenkins restart.
-That is, while the pipeline is running your script, you can shut down Jenkins or lose connectivity to an agent. 
-When it comes back, Jenkins remembers what it was doing and your pipeline script resumes execution as if it were never interrupted.
+Pipeline code is CPS-transformed so that Pipelines are able to resume after a Jenkins restart. That is, while the pipeline is running your script, you can shut down Jenkins or lose connectivity to an agent. When it comes back, Jenkins remembers what it was doing and your pipeline script resumes execution as if it were never interrupted.
```

Comment:
```suggestion
Pipeline code is CPS-transformed so that Pipelines are able to resume after a Jenkins restart.
That is, while the pipeline is running your script, you can shut down Jenkins or lose connectivity to an agent.
When it comes back, Jenkins remembers what it was doing and your pipeline script resumes execution as if it were never interrupted.
```

And so on - better for future reviews and modifications